src/import-utils.ts

```diff
@@ -0,0 +1,301 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import * as path from 'path';
+import * as urlLib from 'url';
+const pathPosix = path.posix;
+import * as dom5 from 'dom5';
+import encodeString from './third_party/UglifyJS2/encode-string';
+
+import * as parse5 from 'parse5';
+import {ASTNode} from 'parse5';
+import {Analyzer, Options as AnalyzerOptions} from 'polymer-analyzer';
+import {Document, ScannedDocument} from 'polymer-analyzer/lib/model/document';
+import {Import} from 'polymer-analyzer/lib/model/import';
+import {ParsedHtmlDocument} from 'polymer-analyzer/lib/html/html-document';
+import {FSUrlLoader} from 'polymer-analyzer/lib/url-loader/fs-url-loader';
+import constants from './constants';
+import * as astUtils from './ast-utils';
+import * as matchers from './matchers';
+import * as urlUtils from './url-utils';
+import {Bundle, BundleStrategy, AssignedBundle, generateBundles, BundleUrlMapper, BundleManifest, sharedBundleUrlMapper, generateSharedDepsMergeStrategy} from './bundle-manifest';
+import {BundledDocument, DocumentCollection} from './document-collection';
+import {buildDepsIndex} from './deps-index';
+import {UrlString} from './url-utils';
+
+// TODO(usergenic): Want to figure out a way to get rid of the basePath param
```

Comment:
Getting rid of basePath helped.  Anyways this comment is changed now to speculate about a class for the import utils to deal with state/shared params etc.